streaming/src/main/scala/io/gearpump/streaming/task/Subscription.scala

```diff
@@ -189,13 +185,38 @@ class Subscription(
     maxPendingCount < maxPendingMessageCount
   }
 
+  def sendAckRequestOnStallingTime(stallingTime: TimeStamp): Unit = {
+    minClockValue.indices.foreach { i =>
+      if (minClockValue(i) == stallingTime && allowSendingMoreMessages) {
+        sendAckRequest(i)
+      }
+    }
+  }
+
+  private def sendAckRequest(partition: Int): Unit = {
+    // we increment more count for each AckRequest
+    // to throttle the number of unacked AckRequest
+    incrementMessageCount(partition, ACKREQUEST_INCREMENT_COUNT)
+    val targetTask = TaskId(processorId, partition)
+    val ackRequest = AckRequest(taskId, messageCount(partition), sessionId)
+    transport.transport(ackRequest, targetTask)
+  }
+
+  private def incrementMessageCount(partition: Int, count: Int = 1): Unit = {
+    messageCount(partition) = (messageCount(partition) + count).toShort
+    pendingMessageCount(partition) = (pendingMessageCount(partition) + count).toShort
+    updateMaxPendingCount()
+  }
+
   private def updateMaxPendingCount() : Unit = {
     maxPendingCount = Shorts.max(pendingMessageCount: _*)
   }
+
 }
 
 object Subscription {
   //make sure it is smaller than MAX_PENDING_MESSAGE_COUNT
   final val ONE_ACKREQUEST_EVERY_MESSAGE_COUNT = 100
   final val MAX_PENDING_MESSAGE_COUNT = 1000
+  final val ACKREQUEST_INCREMENT_COUNT = 100
```

Comment:
NOTE, this is default value. The actual value used is at: 
```
 class Subscription(
     appId: Int,		     appId: Int,
     executorId: Int,		     executorId: Int,
     taskId: TaskId,		     taskId: TaskId,
     subscriber: Subscriber, sessionId: Int,		     subscriber: Subscriber, sessionId: Int,
     transport: ExpressTransport,		     transport: ExpressTransport,
     maxPendingMessageCount: Int = MAX_PENDING_MESSAGE_COUNT,		     maxPendingMessageCount: Int = MAX_PENDING_MESSAGE_COUNT,
     ackOnceEveryMessageCount: Int = ONE_ACKREQUEST_EVERY_MESSAGE_COUNT) {		     ackOnceEveryMessageCount: Int = ONE_ACKREQUEST_EVERY_MESSAGE_COUNT) {
```

You need to make sure "ACKREQUEST_INCREMENT_COUNT" always match the effective ackOnceEveryMessageCount.

So may be we should just use "ackOnceEveryMessageCount".